Documentation/upgrading.md

```diff
@@ -1,173 +1,149 @@
-
-    $ kubectl get daemonsets -n=kube-system
-    NAME             DESIRED   CURRENT   NODE-SELECTOR                    AGE
-    kube-apiserver   1         1         node-role.kubernetes.io/master   5m
-    kube-proxy       3         3         <none>                           5m
-    kubelet          3         3         <none>                           5m
-
-    $ kubectl get deployments -n=kube-system
-    NAME                      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
-    kube-controller-manager   1         1         1            1           5m
-    kube-dns-v20              1         1         1            1           5m
-    kube-scheduler            1         1         1            1           5m
-
-Check the current Kubernetes version.
-
-    $ kubectl version
-    Client Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.0", GitCommit:"a16c0a7f71a6f93c7e0f222d961f4675cd97a46b", GitTreeState:"clean", BuildDate:"2016-09-26T18:16:57Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
-    Server Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.1+coreos.0", GitCommit:"b7a02f46b972c5211e5c04fdb1d5b86ac16c00eb", GitTreeState:"clean", BuildDate:"2016-10-11T20:13:55Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
-
-In this case, Kubernetes is `v1.4.1+coreos.0` and our goal is to upgrade to `v1.4.3+coreos.0`. First, update the control plane pods. Then the kubelets and proxies on all nodes.
-
-## Control Plane
-
-### kube-apiserver
+bootkube does not provide in-place patch diffs or migrations between releases. In-place upgrades are inherently more dangerous than a blue/green cluster deployments. Evaluate whether an in-place upgrade is appropriate for your Kubernetes cluster and risk tolerance before beginning. Failover important workloads to another cluster in advance.
 
-Edit the kube-apiserver daemonset. Change the container image name to `quay.io/coreos/hyperkube:v1.4.3_coreos.0`.
+## Prepare
 
-    $ kubectl edit daemonset kube-apiserver -n=kube-system
+Find the diff between bootkube assets between the existing cluster version and the desired cluster version. This depends on the tool used to generate assets:
 
-Since daemonsets don't yet support rolling, manually delete each apiserver one by one and wait for each to be re-scheduled.
+* Manual - Check Github [compare](https://github.com/kubernetes-incubator/bootkube/compare/v0.6.0...v0.6.1) between existing and diesred versions and infer the appropriate cluster changes
+* [bootkube render](https://github.com/kubernetes-incubator/bootkube) - install the `bootkube` binaries for the existing and desired versions. Render assets to different locations with each binary and diff the assets.
+* [poseidon/bootkube-terraform](https://github.com/poseidon/bootkube-terraform) - checkout the tag for the existing and desired versions. Run `terraform apply` at each tag to render assets to different locations and diff the assets.
```

Comment:
We probably shouldn't single out particular external tool here. What if we had an entry for:

* [External Integrations](https://github.com/kubernetes-incubator/bootkube/blob/master/Documentation/users-integrations.md) - If you used an external tool to provision your cluster, please see their respective documentation for upgrade instructions.

Reply:
True, matchbox calls only calls these out because these are the two tools used in Matchbox examples, for non-terraform and terraform examples respectively. I ported the docs from there.

Will change.